spring-cloud-gcp-storage/src/main/java/org/springframework/cloud/gcp/storage/GoogleStorageResource.java

```diff
@@ -0,0 +1,332 @@
+	 */
+	public GoogleStorageResource(Storage storage, String location) {
+		this(storage, location, true);
+	}
+
+	public boolean isAutoCreateFiles() {
+		return this.autoCreateFiles;
+	}
+
+	@Override
+	public boolean exists() {
+		try {
+			return isBucket() ? getBucket() != null : getBlob() != null;
+		}
+		catch (IOException e) {
+			return false;
+		}
+	}
+
+	@Override
+	public boolean isReadable() {
+		return !isBucket();
+	}
+
+	@Override
+	public boolean isOpen() {
+		return false;
+	}
+
+	/**
+	 * Since the gs: protocol will normally not have a URL stream handler registered,
+	 * this method will always throw a {@link java.net.MalformedURLException}.
+	 * @return The URL for the GCS resource, if a URL stream handler is registered for the gs protocol.
+	 */
+	@Override
+	public URL getURL() throws IOException {
+		return this.location.toURL();
+	}
+
+	@Override
+	public URI getURI() throws IOException {
+		return this.location;
+	}
+
+	/**
+	 * Gets the underlying storage object in Google Cloud Storage.
+	 * @return The storage object, will be null if it does not exist in Google Cloud Storage.
+	 * @throws IOException
+	 * @throws IllegalStateException if the resource reference is to a bucket, and not a blob.
+	 */
+	public Blob getBlob() throws IOException {
+		return this.storage.get(getBlobId());
+	}
+
+	/**
+	 * Creates a signed URL to an object, if it exists. This method will fail if this storage
+	 * resource was not created using service account credentials.
+	 * @param timeUnit the time unit used to determine how long the URL is valid.
+	 * @param timePeriods the number of periods to determine how long the URL is valid.
+	 * @return the URL if the object exists, and null if it does not.
+	 * @throws IllegalStateException if the resource reference is to a bucket, and not a blob.
+	 */
+	public URL createSignedUrl(TimeUnit timeUnit, long timePeriods)  {
+		if (LOGGER.isWarnEnabled()) {
+			if (!exists()) {
+				LOGGER.warn("Creating signed URL for non-existing GCS object " + this.location);
+			}
+		}
+
+		return this.storage.signUrl(
+				BlobInfo.newBuilder(getBlobId()).build(), timePeriods, timeUnit);
+	}
+
+	/**
+	 * Creates the blob that this {@link GoogleStorageResource} represents in Google Cloud
+	 * Storage.
+	 * @return the created blob object
+	 * @throws StorageException if any errors during blob creation arise,
+	 * such as if the blob already exists
+	 * @throws IllegalStateException if the resource reference is to a bucket, and not a blob.
+	 */
+	public Blob createBlob() throws StorageException {
+		return this.storage.create(BlobInfo.newBuilder(getBlobId()).build());
+	}
+
+	/**
+	 * Creates the bucket that this resource references in Google Cloud Storage.
+	 * @return the {@link Bucket} object for the bucket
+	 * @throws StorageException if any errors during bucket creation arise,
+	 * such as if the bucket already exists
+	 */
+	public Bucket createBucket() {
```

Comment:
It doesn't look like this method is used. Didn't you mention not wanting to auto-create buckets?

Reply:
It a public API to allow users to explicitly create the bucket.